src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTaskTest.java

```diff
@@ -123,4 +126,23 @@ public void should_fail_to_upload_to_sftp_and_stop_from_uploading_any_other_lett
             assertThat(l.getFileContent()).isNotNull();
         }
     }
+
+    @Test
+    public void should_process_all_letter_batches() throws Exception {
+        IntStream.rangeClosed(1, UploadLettersTask.BATCH_SIZE * 2).forEach(
+            x -> letterService.send(SampleData.letterRequest(), "service"));
+
+        // and
+        UploadLettersTask task = new UploadLettersTask(
+            repository,
+            FtpHelper.getSuccessfulClient(LocalSftpServer.port),
+            availabilityChecker
+        );
+
+        try (LocalSftpServer server = LocalSftpServer.create()) {
+            task.run();
+        }
+        entityManager.clear();
```

Comment:
may be worth adding `// Clear the JPA cache to force a read.` here as well. (or perhaps extract to a method?)